stock_request/models/stock_request.py

```diff
@@ -21,12 +34,7 @@ def _get_expected_date():
 
     name = fields.Char(states={"draft": [("readonly", False)]})
     state = fields.Selection(
-        selection=[
-            ("draft", "Draft"),
-            ("open", "In progress"),
-            ("done", "Done"),
-            ("cancel", "Cancelled"),
-        ],
+        selection=_get_request_states,
```

Comment:
Prefer a lambda here to help override and put fileds definition on top

Reply:
Please don't. No need of that to insert elements where you want. See https://github.com/OCA/stock-logistics-warehouse/blob/4e338d152d79d99e298ccf34f548515a3a7aa8f7/stock_request_submit/models/stock_request.py#L11

and if you use a method, the translations are very bad handled.